include/mxnet/storage.h

```diff
@@ -26,15 +26,52 @@
 #define MXNET_STORAGE_H_
 
 #include <memory>
+#include <string>
 #include "./base.h"
 
 namespace mxnet {
 
+namespace {
+/// \brief Given a path, extract the filename.
+inline std::string __extract_fname(const std::string& path) {
+  std::size_t last_dir_pos = path.find_last_of("/\\");
+  if (last_dir_pos == std::string::npos) {
+    return path;
+  }
+  return path.substr(last_dir_pos + 1);
+}
+}  // namespace anonymous
+
+#if __GNUG__  // if compiled with GCC
+#define MXNET_STORAGE_DEFAULT_NAME_FARG(tag) \
+    std::string(tag) \
+    + "_" + __extract_fname(__FILE__) \
+    + "+" +  std::to_string(__LINE__) \
+    + "_" + __extract_fname(__builtin_FILE()) \
+    + "+" +  std::to_string(__builtin_LINE())
+#else  // !__GNUG__
+#define MXNET_STORAGE_DEFAULT_NAME_FARG(tag) \
+    std::string(tag) \
+    + "_" + __extract_fname(__FILE__) \
+    + "+" +  std::to_string(__LINE__)
+#endif  // __GNUG__
+#define MXNET_STORAGE_DEFAULT_PROFILER_SCOPE_CSTR  "<unk>:"
+
 /*!
  * \brief Storage manager across multiple devices.
  */
 class Storage {
  public:
+  enum class DataStruct {
```

Comment:
this name seems too generic. It looks like this enum differentiates the type of data allocation. Shall we think of a more informative name in the line of that?

Reply:
Let me just remove this. Since `profiler_scope` contains information regarding the purpose of allocation.